mlir/lib/Dialect/Linalg/IR/LinalgOps.cpp

```diff
@@ -2292,9 +2292,40 @@ Speculation::Speculatability BroadcastOp::getSpeculatability() {
   return getGenericSpeculatabilityImpl(cast<LinalgOp>(getOperation()));
 }
 
+/// Fold broadcast with broadcast.
+struct FoldBroadcastWithBroadcast : OpRewritePattern<linalg::BroadcastOp> {
```

Comment:
Nit: I'd call it `FoldBroadcasts` (plural). The redundancy doesn't help understanding.

I'd also describe the comment with "fold back-to-back broadcasts together".